packages/js/settings-editor/src/legacy/form.tsx

```diff
@@ -50,30 +60,60 @@ export const Form = ( {
 		return formData;
 	};
 
-	const handleSubmit = ( event: React.FormEvent< HTMLFormElement > ) => {
+	const handleSubmit = async (
+		event: React.FormEvent< HTMLFormElement >
+	) => {
 		event.preventDefault();
+		setIsBusy( true );
 
 		const query: Record< string, string > = {
 			page: 'wc-settings',
-			tab: settingsPage.slug,
 		};
+		if ( settingsPage.slug !== 'general' ) {
+			query.tab = settingsPage.slug;
+		}
 		if ( activeSection !== 'default' ) {
 			query.section = activeSection;
 		}
 
-		const formData = getFormData();
-		formData.save = 'Save changes';
-		formData._wpnonce = settingsData._wpnonce;
-		formData._w_http_referer = '/wp-admin/' + getNewPath( query );
-
-		// eslint-disable-next-line no-console
-		console.log(
-			'tab: ',
-			settingsPage.slug,
-			'section: ',
-			activeSection,
-			formData
+		const updatedData = getFormData();
+		updatedData.save = 'Save changes';
+		updatedData._wpnonce = settingsData._wpnonce;
+		updatedData._w_http_referer = '/wp-admin/' + getNewPath( query );
+
+		const payload = new FormData();
+		for ( const [ key, value ] of Object.entries( updatedData ) ) {
+			payload.append( key, value );
+		}
+
+		const response = await fetch(
+			`/?rest_route=/wc-admin/legacy-settings&${ new URLSearchParams(
+				query
+			).toString() }`,
+			{
+				method: 'POST',
+				body: payload,
+				credentials: 'same-origin', // Include cookies for nonce validation
+				headers: {
+					'X-WP-Nonce': settingsData._wpnonce,
+				},
+			}
 		);
+
+		const responseData = await response.json();
```

Comment:
This might throw an error. I think we need to add a `try...catch`.

```ts
    try {
        // ... existing code ...
    } catch (error) {
        createNotice(
            'error',
            __('Failed to save settings. Please try again.', 'woocommerce')
        );
        console.error('Settings save error:', error);
    } finally {
        setIsBusy(false);
    }
```

Reply:
Fixed in [add try/catch to response handling](https://github.com/woocommerce/woocommerce/pull/55467/commits/8925a710987d1551ec6a3e18d5cd330198326f2e)